### PR TITLE
feat(search-form): migrate component search form - FRONT-1775

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

migrate component search form to new structure

### CODE REVIEW

File: src/implementations/twig/components/search-form/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+### Parameters
+
+- **"text_input"** (associative array) default: A predefined structure for EC Text Input
+- **"button"** (associative array) default: A predefined structure for EC Button
+- **"extra_form_elements"** (optional) (string) (default: '') Extra elements for the form
+- **"extra_classes"** (optional) (string) (default: '') Extra classes (space separated)
+- **"extra_attributes"** (optional) (array) (default: []) Extra attributes
+  - "name" (string) Attribute name, eg. 'data-test'
+  - "value" (string) Attribute value, eg: 'data-test-1'
+- **"\_compliance"** (boolean) (default: false) Activates debug
```
Comment (planctus) [inline src/implementations/twig/components/search-form/README.md]:
```suggestion
- **"\_compliance\_"** (boolean) (default: false) Activates debug
```

File: src/implementations/twig/components/search-form/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+- **"_compliance_inner_check_"** (boolean) (default: false) Inline compliance report
```
Comment (planctus) [inline src/implementations/twig/components/search-form/README.md]:
```suggestion
- **"\_compliance_inner_check\_"** (boolean) (default: false) Inline compliance report
```

File: src/implementations/twig/components/search-form/__snapshots__/search-form.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+  <input
+    id="custId"
+    name="custId"
+    type="hidden"
+    value="1"
+  />
+</form>
+`;
+
+exports[`Search Form with validation enabled and missing input data returns the right warning message 1`] = `
```
Comment (planctus) [inline src/implementations/twig/components/search-form/__snapshots__/search-form.test.js.snap:207]: Not sure it's worth checking this, but i don't see any output from the compliance tool here

File: src/implementations/twig/components/search-form/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+npm package: `@ecl/twig-component-search-form`
+
+```shell
+npm install --save @ecl/twig-component-search-form
+```
+
+### Parameters
+
+- **"text_input"** (associative array) default: A predefined structure for EC Text Input
```
Comment (planctus) [inline src/implementations/twig/components/search-form/README.md]:
```suggestion
- **"text_input"** (associative array) default: Predefined structure for the text-Input component
```

File: src/implementations/twig/components/search-form/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+- **"button"** (associative array) default: A predefined structure for EC Button
```
Comment (planctus) [inline src/implementations/twig/components/search-form/README.md]:
```suggestion
- **"button"** (associative array) default: Predefined structure for the Button component
```

**Review by planctus — CHANGES_REQUESTED** [review]
Super small things, as usual.. :)

File: src/implementations/twig/components/search-form/__snapshots__/search-form.test.js.snap (hunk 3 above)
Comment (emeryro) [reply src/implementations/twig/components/search-form/__snapshots__/search-form.test.js.snap:207]: honnestly, I didn't really check what test is run here.
As we don't know if we will keep the compliance tool as is in the end, I'd say that we can handle this at a later stage

File: src/implementations/twig/components/search-form/search-form.story.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  getComplianceKnob(data);
+
+  return data;
+};
+
+export default {
+  title: 'Components/Forms/Search Form',
+};
+
+export const Default = () => searchForm(prepareSearchForm(dataDefault));
```
Comment (kalinchernev) [inline src/implementations/twig/components/search-form/search-form.story.js:35]: Strangely, https://ecl-preview-1760--europa-component-library.netlify.app/playground/ec/?path=/story/components-forms-search-form--default does not seem to look like https://ecl-twig-js.netlify.app/ec/?path=/story/components-forms-search-form--default 

The search label doesn't seem to show up. Is it a desired behavior? If yes, we can remove the knob i think.

**Review by kalinchernev — COMMENTED** [review]
Only 1 question from me on this one

**Review by kalinchernev — APPROVED** [review]

File: src/implementations/twig/components/search-form/search-form.story.js (hunk 6 above)
Comment (kalinchernev) [reply src/implementations/twig/components/search-form/search-form.story.js:35]: It was the difference in addons panel. So the panel of responsive was below the breakpoint to get the same style.
Bref, no issues.

**Pull request merged**